tests/hpc/slurm_master_db.pm

```diff
@@ -18,6 +18,165 @@ use warnings;
+}
+
+sub validate_result {
+    my ($result) = @_;
+
+    if ($result == 0) {
+        return 'PASS';
+    } elsif ($result == 1) {
+        return 'FAIL';
+    } else {
+        return undef;
+    }
+}
+
+sub results {
+    my ($description, $result) = @_;
+
+    my %results = (
+        test        => get_test_name(),
+        description => $description,
+        result      => validate_result($result)
+    );
+    return %results;
+}
+
+sub pars_results {
+    my (%test)  = @_;
+    my $file    = 'tmpresults.xml';
+    my @results = keys %test;
+
+    my $test_name;
+    my $description;
+    my $result;
+
+    for my $i (@results) {
+        if ($i eq 'test') {
+            $test_name = $test{$i};
+        } elsif ($i eq 'description') {
+            $description = $test{$i};
+        } elsif ($i eq 'result') {
+            if ($test{$i} eq 'FAIL') {
+                $result = $test{$i};
+            }
+        } else {
+            die('Wrong key!');
+        }
+    }
+
+    if ($result eq 'FAIL') {
+        script_run("echo \"<testcase name='$test_name' errors='1'>\" | tee -a $file");
+    } else {
+        script_run("echo \"<testcase name='$test_name'>\" | tee -a $file");
+    }
+    script_run("echo \"<system-out>\" | tee -a $file");
+    script_run("echo $description | tee -a $file");
+    script_run("echo \"</system-out>\" | tee -a $file");
+    script_run("echo \"</testcase>\" | tee -a $file");
+}
+
+sub test_01 {
+    my $description = 'Basic SRUN test';
+    my $result;
+
+    $result = script_run("srun -w slave-node00 date");
+
+    my %results = results($description, $result);
+    return %results;
+}
+
+sub test_02 {
+    my $description = 'Simple SINFO test';
+    my $result;
+
+    $result = script_run('sinfo');
+
+    ##TODO: add check of sinfo vs slurm.conf?
+
+    my %results = results($description, $result);
+    return %results;
+}
+
+##TODO: improve before enabling
+sub test_03 {
+    my $description = 'SRUN stress test';
+    my $result      = 0;
+    my $result0;
+    my $result1;
+
+    my @collected_results;
+
+    for (my $i = 0; $i < 50; $i++) {
+        $result0 = script_run("srun -w slave-node00 date");
+        push(@collected_results, $result0);
+        $result1 = script_run("srun -w slave-node01 date");
+        push(@collected_results, $result1);
+    }
+
+    foreach my $i (@collected_results) {
+        if ($i == 1) {
+            $result = 1;
+        }
+    }
+
+    my %results = results($description, $result);
+    return %results;
+}
+
+sub test_04 {
+    my $description = 'Basic SBATCH test';
+    my $sbatch      = 'slurm_sbatch.sh';
+    my $result;
+
+    script_run("wget --quiet " . data_url("hpc/$sbatch") . " -O $sbatch");
+    assert_script_run("chmod +x $sbatch");
+    record_info('meminfo', script_output("cat /proc/meminfo"));
+    $result = script_output("sbatch $sbatch");
+    ##sbatch SBATCH --time=0-00:01:00
+    ## so the worker should wait for the sbatch to finish
+    sleep(70);
```

Comment:
needed? is sbatch executed in background? how? script_output will hang until command finishes. 
And if so, as a suggestion, would it be a better approach to check for the process, instead of sleep?

Reply:
I was just curious, https://slurm.schedmd.com/sbatch.html
```
sbatch exits immediately after the script is successfully transferred to the Slurm controller and assigned a Slurm job ID.
```

Reply:
Ok. So improvement here is to check for the process if it's running or not.

Reply:
without having any deeper detail, maybe `--wait` will do it as well?

Reply:
will check it.

Reply:
ok for me to leave as it is and improve it later on, but worth to look at that `--wait`